salt/engines/slack.py

```diff
@@ -135,6 +146,25 @@ def fire(tag, msg):
                                 # Trim the ! from the front
                                 # cmdline = _text[1:].split(' ', 1)
                                 cmdline = salt.utils.shlex_split(_text[len(trigger):])
+
+                                # Remove inital salt item from list
+                                if cmdline[0] == 'salt':
+                                    del cmdline[0]
+
+                                # Remove slack url parsing
+                                #  Translate target=<http://host.domain.net|host.domain.net>
+                                #  to target=host.domain.net
+                                cmdlist = []
+                                for cmditem  in cmdline:
```

Comment:
If you have minions named hostname, all is fine. If any are named in the format hostname.domain, slack auto-parses that into a url server side and this becomes important.